src/sentry/auth/helper.py

```diff
@@ -224,60 +225,41 @@ def _handle_new_membership(self, auth_identity):
 
         return om
 
+    def _get_auth_identity(self, **params) -> Optional[AuthIdentity]:
+        try:
+            return AuthIdentity.objects.get(auth_provider=self.auth_provider, **params)
+        except AuthIdentity.DoesNotExist:
+            return None
+
     @transaction.atomic
     def handle_attach_identity(self, identity, member=None):
         """
         Given an already authenticated user, attach or re-attach an identity.
         """
-        try:
-            try:
-                # prioritize identifying by the SSO provider's user ID
-                auth_identity = AuthIdentity.objects.get(
-                    auth_provider=self.auth_provider, ident=identity["id"]
-                )
-            except AuthIdentity.DoesNotExist:
-                # otherwise look for an already attached identity
-                # this can happen if the SSO provider's internal ID changes
-                auth_identity = AuthIdentity.objects.get(
-                    auth_provider=self.auth_provider, user=self.user
-                )
-        except AuthIdentity.DoesNotExist:
+        # prioritize identifying by the SSO provider's user ID
+        auth_identity = self._get_auth_identity(ident=identity["id"])
+        if auth_identity is None:
+            # otherwise look for an already attached identity
+            # this can happen if the SSO provider's internal ID changes
+            auth_identity = self._get_auth_identity(user=self.user)
+
+        auth_is_new = auth_identity is None
+        if auth_is_new:
```

Comment:
i kind of like checking `if auth_identity is None:` to keep similar as above and setting the `auth_is_new` variable within the block, this kind of confused me looking at it at first. your call though

Reply:
Exactly the kind of thing on which I want subjective feedback. 😉 Let's go with your way.